src/Uplink/Admin/Ajax.php

```diff
@@ -22,34 +22,47 @@ public function __construct() {
 	 * @since 1.0.0
 	 * @return void
 	 */
-	public function validate_license() {
+	public function validate_license(): void {
 		$submission = [
-			'_wpnonce' => isset( $_POST['_wpnonce'] ) ? sanitize_text_field( wp_unslash( $_POST['_wpnonce'] ) ) : '',
-			'plugin'   => isset( $_POST['plugin'] ) ? sanitize_text_field( wp_unslash( $_POST['plugin'] ) ) : '',
-			'key'      => isset( $_POST['key'] ) ? Utils\Sanitize::key( wp_unslash( $_POST['key'] ) ) : '',
+			'_wpnonce' => sanitize_text_field( wp_unslash( $_POST['_wpnonce'] ?? '' ) ),
+			'plugin'   => sanitize_text_field( wp_unslash( $_POST['plugin'] ?? '' ) ),
+			'key'      => Utils\Sanitize::key( wp_unslash( $_POST['key'] ?? '' ) ),
 		];
 
 		if ( empty( $submission['key'] ) || ! wp_verify_nonce( $submission['_wpnonce'], $this->container->get( License_Field::class )->get_group_name() ) ) {
-			echo json_encode( [
+			wp_send_json_error( [
 				'status'  => 0,
-				'message' => __( 'Invalid request: nonce field is expired. Please try again.', '%TEXTDOMAIN%' )
+				'message' => __( 'Invalid request: nonce field is expired. Please try again.', '%TEXTDOMAIN%' ),
 			] );
-			wp_die();
 		}
 
 		$collection = $this->container->get( Collection::class );
-		$plugin     = $collection->current();
+		$plugins    = $collection->get_by_path( $submission['plugin'] );
```

Comment:
@borkweb  This can potentially return multiple plugins with the same, but we're only accepting a single license key above, and using that to validate against all resources, is that going to be a problem?

Reply:
Done